tests/unit/test_row_noise.py

```diff
@@ -114,6 +115,134 @@ def test_do_not_respond_missing_columns(dummy_data):
+            "housing_type": [
+                "Household",
+                "Household",
+                "Household",
+                "College",
+                "College",
+                "Household",
+                "Household",
+                "Household",
+                "Household",
+                "Household",
+            ],
+            "age": [10, 10, 10, 20, 20, 20, 50, 50, 50, 50],
+            "guardian_1": ["8", "7", "5", "9", "7", "6", np.nan, np.nan, np.nan, np.nan],
+            "guardian_2": [
+                "9",
+                np.nan,
+                np.nan,
+                "8",
+                np.nan,
+                np.nan,
+                np.nan,
+                np.nan,
+                np.nan,
+                np.nan,
+            ],
+            "relationship_to_reference_person": ["Fake value"] * 10,
+            "year": [2019] * 10,
+            "street_name": [
+                "Street 0",
+                "Street 0",
+                "Street 1",
+                "Street 1",
+                "Street 1",
+                "Street 2",
+                "Street 3",
+                "Street 3",
+                "Street 4",
+                "Street 5",
+            ],
+            "street_number": [0, 0, 1, 1, 1, 2, 3, 3, 4, 5],
+            "unit_number": [0, 0, 1, 1, 1, 2, 3, 3, 4, 5],
+            "city": [
+                "City 0",
+                "City 0",
+                "City 1",
+                "City 1",
+                "City 1",
+                "City 2",
+                "City 3",
+                "City 3",
+                "City 4",
+                "City 5",
+            ],
+            "state": [
+                "State 0",
+                "State 0",
+                "State 1",
+                "State 1",
+                "State 1",
+                "State 2",
+                "State 3",
+                "State 3",
+                "State 4",
+                "State 5",
+            ],
+            "zipcode": [
+                00000,
+                00000,
+                11111,
+                11111,
+                11111,
+                22222,
+                33333,
+                33333,
+                44444,
+                55555,
+            ],
+        }
+    )
+    # Noise 100% of rows
+    overrides = {
+        key: 1.0
+        for key in get_configuration()[DATASETS.census.name][Keys.ROW_NOISE][
+            NOISE_TYPES.duplicate_with_guardian.name
+        ]
+    }
+    dataset_name_1 = DATASETS.census.name
+    noised = NOISE_TYPES.duplicate_with_guardian(
+        dataset_name_1, dummy_data, overrides, RANDOMNESS
+    )
+
+    # We know the following since every dependent is duplicated:
+    #  - Simulant ids 0-5 will all be duplicated
+    #  - Simulant ids 5-9 are guardians. The only overlap is simulant id 5,
+    #    who is both a dependent and a guardian
+    #  - Simulant id 0and 4 have two guardians, 8 and 9
```

Comment:
type: space between 0 and "and"

Reply:
I dont like that as much because I read that as 0 through 4

Reply:
oh yeah, you're totally right